frontend/packages/console-plugin-sdk/src/codegen/plugin-resolver.ts

```diff
@@ -56,11 +59,25 @@ export const filterActivePluginPackages: PluginPackageFilter = (appPackage, plug
   );
 };
 
+/**
+ * Resolve Console monorepo root directory by traversing `package.json` files.
+ */
+export const getMonorepoRootDir = () => {
+  return findUp.sync(
+    (currentDir) => {
+      return fs.existsSync(path.join(currentDir, 'package.json'))
+        ? readPkg.sync({ cwd: currentDir, normalize: true }).name === '' && currentDir
+        : undefined;
+    },
+    { cwd: __dirname, type: 'directory' },
```

Comment:
Note: using `__dirname` here, instead of e.g. `process.cwd()`, for more reliable detection.